include/cinder/Log.h

```diff
@@ -135,18 +140,46 @@ class LoggerBreakpoint : public Logger {
 	Level	mTriggerLevel;
 };
 
-#if defined( CINDER_COCOA )
-
-class LoggerSysLog : public Logger {
+class LoggerSystem : public Logger {
 public:
-	LoggerSysLog();
-	virtual ~LoggerSysLog();
-
-	void write( const Metadata &meta, const std::string &text ) override;
-};
-
+	LoggerSystem();
+	virtual ~LoggerSystem();
+	
+	void write( const Metadata &meta, const std::string &text ) override { mImpl->write( meta, text ); }
+	// SysLog and EventLog logging levels inherently work differently (see specific comments).  Currently
+	// this doesn't matter since logging implimentation only allows once instance of a logger type.
```

Comment:
Sorry completely missed this comment somehow.




I think the pimpl makes a lot of sense and I implemented it. 




I gave a lot of thought to the LoggerSystem filtering. I think we need to explore optimized methods for setting logging filters down the road, but for now I like consistency between platforms and I doubt setlogmask can be any more efficient than a bool check. I also didn't love how it has a global application effect if you call it but use syslog elsewhere - but I'm not sure that's a real use case anyway. 




I think we need secondary control over system log levels because I see them used for reporting warnings or errors not general debug logging. There are tons of tools that scan event logs or syslogs and report on errors and they are what I think sys logging is good for. Other logs should just append to the (rotating) log file.

On Sat, Jun 6, 2015 at 11:08 PM, Richard Eakin <notifications@github.com>
wrote:

>>  public:
>> -	LoggerSysLog();
>> -	virtual ~LoggerSysLog();
>> -
>> -	void write( const Metadata &meta, const std::string &text ) override;
>> -};
>> -
>> +	LoggerSystem();
>> +	virtual ~LoggerSystem();
>> +	
>> +	void write( const Metadata &meta, const std::string &text ) override { mImpl->write( meta, text ); }
>> +	// SysLog and EventLog logging levels inherently work differently (see specific comments).  Currently
>> +	// this doesn't matter since logging implimentation only allows once instance of a logger type.
> Ah, yes I see, they are quite different. Essentially what you've done for EventLog is add runtime level filtering support. In the future I'd like to add this at a more generic level, perhaps the LoggerMulti, so we get the functionality for even things like LoggerFile, but I'd like to do it so it is minimal overhead when turned off - that is the tricky part.
> It is a tough decision, what to do around ::setlogmask(). Part of me thinks it should be a static LoggerSystem::setLogMask(), that is only available on platforms where syslog exists. If that were done, it would make sense to either move the EventLog's runtime level filtering to LoggerSystem, or remove it altogether, since it will likely come back in a more generic form. 
> Any thoughts around the pimpl move? I can get this merged and do the change locally if you prefer, or let me know if you have other opinions. I'd like to get this PR closed soon since it is against glNext, which is going to be removed soon since it was merged into master branch.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cinder/Cinder/pull/921/files#r31873741